executor.go

```diff
@@ -420,6 +426,61 @@ func executeShell(shellName string, command string, env []string, stage string,
 	defer cancel()
 
 	cmd := exec.CommandContext(ctx, shellName, f.Name())
+
+
+	cmd.Env = append(os.Environ(), env...)
+
+	output, err := cmd.CombinedOutput()
+	if err != nil {
+		if context.DeadlineExceeded == ctx.Err() {
+			return string(output), fmt.Errorf("TIMED OUT: script %w", err)
+		}
+		return string(output), fmt.Errorf("executing %s script: %w", shellName, err)
+	}
+/*
+	err = ctx.Err()
```

Comment:
please back-out the changes to the `executeShell()` function, now that you have added executePS and executeCMD